src/open_inwoner/openzaak/info_objects.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from dataclasses import dataclass
+from datetime import date, datetime
+from typing import List, Optional
+
+import requests
+from requests import HTTPError, RequestException, Response
+from zds_client import ClientError
+from zgw_consumers.api_models.base import ZGWModel, factory
+
+from open_inwoner.openzaak.clients import build_client
+from open_inwoner.openzaak.statuses import logger
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class InformatieObject(ZGWModel):
```

Comment:
@annashamray  This was a change necessary for all due to the zgw_consumers models not being compatible with the e-Suite ZGW API, as documented here: https://taiga.maykinmedia.nl/project/open-inwoner/wiki/afwijkingen-e-suite-tov-zgw-standaard

In short, e-Suite ZGW API doesn't return all the keys in responses which zgw_consumers assumes does always return. This was at least the case this summer.

Reply:
Oh! I completely forgot (and maybe didn't know) about e-Suite
But taiga link includes only `ZaakInformatieObject` differences, it doesn't include any info about `InformatieObject` resource

So I'll keep this comment unresolved, but resolve the same comment about `ZaakInformatieObject` class

Reply:
You're correct, we didn't request IO's earlier, only ZIO's. 

I expect we will also need to tweak the IO model further to get this to work with e-Suite, so having an OIP-specific IO model makes this more workable. For this current iteration we can keep the scope to only integrating with Open Zaak, once this is on staging we can do further e-Suite testing to double-check if this also works there as expected.

The alternative is that we temporarily deploy this branch on acceptance.

Reply:
Does e-Suite just doesn't add certain fields or do they have different fields altogether?

I think we should look into fixing the `factory()` function so it at least doesn't crash when fields are missing (it did that even for the IO model from ZGW and the test OpenZaak environment). Or we can keep two sets of dataclasses, one of OpenZaak and one for e-Suite.

Reply:
I discussed with @annashamray and we leave it like this for now to land this PR, and then create an issue look at refactoring the factory function so we can run e-Suite and OpenZaak from same dataclasses.

Reply:
e-Suite adheres to the same JSON responses as OpenZaak, however with less fields. But the main difference is that empty fields aren't returned in the JSON response.

If you want we can investigate this on staging together if necessary based on the acceptance environments.

Reply:
If it is just less fields then we should be able to improve the factory function to deal with that and set some default/empty values.

Reply:
@sergei-maertens noted we can easily create E-Suite compatible dataclasses by extending the official dataclasses and overriding the problematic missing fields with fields with Optional datatype and a default value.

This would keep us from compromising our accuracy for their non-spec implementation and make the differences visible in the E-suite compatible dataclasses.

We should pick that up in a different task once this PR lands.

Reply:
I created a task for this: https://taiga.maykinmedia.nl/project/open-inwoner/task/920